draft-ietf-cdni-uri-signing.xml

```diff
@@ -544,6 +544,33 @@
             claim if no Client IP claim existed in the received
             signed JWT.</t>
         </section>
+        <section anchor="cdnib_claim" title="CDNI Base (cdnib) claim">
+            <t>Base (cdnir) [optional] -
+            The base path for which the token is intended. Its type is a
+            JSON string. It contains a <xref target="RFC3986">URI</xref>,
+            except that any or all of the scheme, host, or port MAY be
+            omitted. This URI MUST NOT contain User Information
+            (<xref target="RFC3986" /> Section 3.2.1), Query (Section 3.4),
+            or Fragment (Section 3.5) components.</t>
+
+            <t>The scheme is terminated by a colon (':'), which MUST be
+            omitted if the scheme is. The authority is preceeded by a
+            double slash ("//"), which MUST be omitted if the authority
+            is, and terminated by the next slash ("/"), which begins the
+            required path, or the end of the URI.</t>
```

Comment:
It feels like we are repeating work here. Isn't this already covered by another claim? Can't we just have this as a path?

I also feel like "Base" is too generic of a name. "Base Path" would be more descriptive perhaps?

Reply:
Unfortunately, no. I'd love to have it as an actual, honest-to-goodness URI. But it would be really useful to omit components. If we don't omit components, we have to invent a glob to wildcard them and all sorts of other silliness. In no event can an implementer use an ordinary URI parsing library for it. Omitting them is as easy to parse as anything else, cause they are each fairly easily delimited. All the text is clarifying what "omitted" means so that we don't wind up with parsers that implement it differently.

Reply:
Though... it should say, "which MUST be omitted if all of the authority is", since it's possible to omit only part of the authority.